test/gtest/uct/ib/test_cqe_zipping.cc

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+#include <common/test_helpers.h>
+#include <uct/uct_test.h>
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/ib/base/ib_iface.h>
+}
+
+#include <utility>
+#include <memory>
+#include <array>
+
+#if defined(ENABLE_STATS)
+#define UCT_INSTANTIATE_MLX5_TEST_CASE(_test_case) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, rc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, dc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, ud_mlx5)
+
+struct test_cqe_zipping : public uct_test {
+    constexpr static size_t buf_size = 4096;
+
+    struct peer {
+        peer(test_cqe_zipping &test) :
+            m_test(test),
+            m_entity(test.create_entity(0)),
+            m_buf(new mapped_buffer(buf_size, 0, *m_entity))
+        {
+            m_test.m_entities.push_back(m_entity);
+            uct_iface_set_am_handler(m_entity->iface(), 0, am_cb, &m_test, 0);
+        }
+
+        test_cqe_zipping               &m_test;
+        entity                         *m_entity;
+        std::unique_ptr<mapped_buffer> m_buf;
+    };
+
+    void send_while_possible()
+    {
+        ucs_time_t deadline = ucs::get_deadline(30);
+        for (auto &peer : peers) {
```

Comment:
As I understand the logic of the test, making a round-robin mae even reduce the probability of CQE zipping since during the round robin we will send messages to different QPs one after another and therefore the time between two messages send to one QP will increase. IRIC HW counts some deadline before placing CQE without zipping, so I don't think it is a good idea.